FreeRTOS_Sockets.c

```diff
@@ -387,6 +387,24 @@ Socket_t FreeRTOS_socket( BaseType_t xDomain,
     EventGroupHandle_t xEventGroup;
     Socket_t xReturn;
 
+    /* A protocol of 0 indicates to the socket layer that it should pick a 
+     * sensible default protocol based off the given socket type. If we can't,
+     * prvDetermineSocketSize will catch it as an invalid type/protocol combo.
+     */
+    if( xProtocol == 0 )
```

Comment:
As an 'aside'. Maybe we should convert 0 to a Macro?

Reply:
While I'm also not a huge fan of magic numbers, the POSIX description of socket() explicitly mentions the numeric value 0. Linux defines IPPROTO_IP as 0 for this purpose, but I don't find that name particularly obvious. Still, I'll gladly patch it in if you want.

Reply:
Yes, I understand. Users will get to use 0 directly. But while someone is reading the code, one might wonder what `0` is about if they are not familiar with 0 being used in POSIX sockets (as I wasn't either). 
But generally macros look better. Maybe we should come up with a good descriptive name for the macro. Something like `FREERTOS_SOCK_DEPENDENT_PROTO`.

Reply:
I went ahead and used your suggestion for a macro name :)